CHANGELOG.md

```diff
@@ -38,6 +38,10 @@
  * Note that `email_verified == not email_verification_required`.
  * A data migration will be necessary.
 
+## v8.1.1
+
+* Bugfix: Don't show "passwords do not match" when the first password is invalid.
```

Comment:
The code presumably got changed at some point between 8.x and 12.x as part of larger works.  Whether to deliberately fix this bug or not, I don't know.